lib/middleware/validateSession.js

```diff
@@ -0,0 +1,70 @@
+var getSessionVal = function(obj, key) {
+  key = key.split('.');
+
+  for (var i = 0; i < key.length; i++) {
+    var curKey = key[i];
+
+    if (curKey in obj) {
+      obj = obj[curKey];
+    } else {
+      return null;
+    }
+  }
+
+  return obj;
+};
+
+module.exports = function validateSession(mediator, mbaasApi, opts) {
+
+  return function validateSessionMiddleware(req, res, next) {
+    var session,
+      sessionToken;
+
+    opts = {
+      sessionKey: opts.sessionKey || "fh_params",
+      sessionIdKey: opts.sessionIdKey || "__fh.sessiontoken",
+      sessionHeaders: opts.sessionHeaders || "x-fh-sessiontoken"
+    };
+
+    session = getSessionVal(req, opts.sessionKey) || {};
+    sessionToken = getSessionVal(session, opts.sessionIdKey) || req.headers[opts.sessionHeaders] || null;
+
+    if (!sessionToken) {
+      console.log("No session token available");
+      return res.status(401).json(new Error("Unauthorized"));
+    }
+
+    //Check cache for session
+    mbaasApi.session.get(sessionToken, function(err, sessionData) {
```

Comment:
This should probably be called `getSessionValidationMiddleware` to be more accurate.